libs/sdk-ui-kit/src/Dialog/ShareDialog/ShareDialogBase/useShareDialogBase.ts

```diff
@@ -146,9 +158,30 @@ export const useShareDialogBase = (props: IShareDialogBaseProps): IUseShareDialo
 
     useGetAccessList({ sharedObjectRef, onSuccess: onLoadGranteesSuccess, onError });
 
+    const [isUnderStrictControl, setUnderStrictControl] = useState(isOriginallyUnderStrictControl);
```

Comment:
Could you pls move handling this states isUnderStrictControl and isLocked in useShareDialogState.

Reply:
Fixed.